pep-9999.rst

```diff
@@ -0,0 +1,883 @@
+  and open source software licenses require to include their full text in a
+  distribution.
+
+These considerations have guided the design and updates proposed in this PEP.
+
+Reusing the existing `license` field with license expressions will provide a an
+intuitive and more structured way to express the license of a distribution using
+a well defined syntax and well known license ids.
+
+Over time, recommending the usage of these expressions will help Python package
+publishers improve the clarity of their license documentation to the benefit of
+packages authors, consumers and redistributors.
+
+
+Core Metadata Specification updates
+===================================
+
+The canonical source for the names and semantics of each of the supported
+metadata fields is the Core Metadata Specification [#cms]_ document.
+
+The details of the updates considered to the Core Metadata Specification [#cms]_
+document as part of his PEP are detailed here and will be added to the
+canonical source once this PEP is approved.
+
+
+Added in Version 2.2
+--------------------
+
+License-File (multiple use)
+:::::::::::::::::::::::::::
+
+The is a string that is a package-root relative path to a license file. The
+license file content __must__ be UTF-8-encoded text.
+
+Build tools should honor this field and include the corresponding license
+file(s) in the built package.
+
+
+Changed in Version 2.2
+----------------------
+
+License (optional)
+::::::::::::::::::
+
+Text indicating the license covering the distribution. This text can be either a
+valid `License Expression` as defined here or any free text.
+
+Publishing tools SHOULD issue an informational warning if this field is empty or
+missing or is not a valid `License Expression` as defined here. Build tools MAY
+issue such a warning too.
+
+
+License Expression syntax
+'''''''''''''''''''''''''
+
+A `License Expression` is a string using the SPDX license expression syntax as
+documented in the SPDX specification [#spdx]_ using either Version 2.1
+[#spdx21]_ or a later compatible version. SPDX is a working group at the Linux
+Foundation that defines a standard way to exchange package information.
+
+When used in the `License` field and as a specialization of the SPDX license
+expression definition, a `License Expression` can use the following license
+identifiers:
+
+- any SPDX-listed license short-form identifiers that are published in the 
+  SPDX License List [#spdxlist]_ using either Version 3.6 of this list or any
+  later compatible version. Note that the SPDX working group never removes any
+  license identifiers: instead they may only one as obsolete.
+
+- the `Public-Domain` and `Proprietary` strings to support generic ids that are
+  not available in the SPDX license list.
+
+When processing the `License` field to determine if it contains a valid license
+expression, tools:
+
+- MUST ignore the case of the `License` field.
```

Comment:
What's the rationale to this?

I mean, the choice is irrelevant as there are no ambiguities.

---

Anecdotally all miss-cases cases I'm aware of were actual mistakes, so IMO one COULD report warning if non-canonical casing is used.

Reply:
https://github.com/spdx/spdx-spec/issues/63 is related issue.

Reply:
> What's the rationale to this?

Accept anything even if there is a case change since this case really does not matter.

> Anecdotally all miss-cases cases I'm aware of were actual mistakes, 

Indeed!

> so IMO one COULD report warning if non-canonical casing is used.

Good point: reporting a warning is a good way.

Reply:
@phadej this has been updated in https://github.com/pombredanne/spdx-pypi-pep/pull/2/commits/d584c484099c3b80bafd0f26ceb1d7a9960e4ba8 to report a warning!
Thank you ++